src/remote/routes.js

```diff
@@ -0,0 +1,242 @@
+
+        reply(makeResponse(id, { initialized: node.initialized }))
+      })
+    }
+  })
+
+  /**
+   * Delete the repo that was being used.
+   * If the node was marked as `disposable` this will be called
+   * automatically when the process is exited.
+   **/
+  server.route({
+    method: 'GET',
+    path: '/shutdown',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      nodes[id].shutdown((err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Start the daemon.
+   **/
+  server.route({
+    method: 'GET',
+    path: '/startDaemon',
+    handler: (request, reply) => {
+      const qr = parseQuery(request.query)
+      const id = qr.id
+      const flags = Object.values(qr.params ? qr.params.flags : {})
+      nodes[id].startDaemon(flags, (err) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, {
+          apiAddr: nodes[id].apiAddr.toString(),
+          gatewayAddr: nodes[id].gatewayAddr.toString()
+        }))
+      })
+    }
+  })
+
+  /**
+   * Stop the daemon.
+   */
+  server.route({
+    method: 'GET',
+    path: '/stopDaemon',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      nodes[id].stopDaemon((err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Kill the `ipfs daemon` process.
+   *
+   * First `SIGTERM` is sent, after 7.5 seconds `SIGKILL` is sent
+   * if the process hasn't exited yet.
+   */
+  server.route({
+    method: 'GET',
+    path: '/killProcess',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      nodes[id].killProcess((err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Get the pid of the `ipfs daemon` process.
+   *
+   * @returns {number}
+   */
+  server.route({
+    method: 'GET',
+    path: '/daemonPid',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      reply(makeResponse(id, nodes[id].daemonPid(nodes[id])))
+    }
+  })
+
+  /**
+   * Call `ipfs config`
+   *
+   * If no `key` is passed, the whole config is returned as an object.
+   */
+  server.route({
+    method: 'GET',
+    path: '/getConfig',
+    handler: (request, reply) => {
+      const qr = parseQuery(request.query)
+      const id = qr.id
+      const key = qr.params ? qr.params.key : null
+      nodes[id].getConfig(key, (err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Set a config value.
+   */
+  server.route({
+    method: 'GET',
+    path: '/setConfig',
```

Comment:
The method here should be a `PUT` and the path should be `/config`.
(Ideally, `GET /config` gets the config, `PUT /config` puts the config).